### PR TITLE
adjustments to model for fundings

### PR DESCRIPTION
**Pull request by ketilaa**

### CODE REVIEW

File: nva-datamodel-java/src/main/java/no/unit/nva/model/Funding.java [hunk 1]
```diff
@@ -10,10 +11,13 @@
...
+    private URI id;
     private String identifier;
-    private String mainTitle;
-    private Map<String, String> alternativeTitles;
-    private MonetaryAmount amount;
+    private String name;
+    private Map<String, String> alternativeName;
+    private MonetaryAmount fundingAmount;
+    private Instant activeFrom;
+    private Instant activeTo;
```
Comment (brinxmat) [inline nva-datamodel-java/src/main/java/no/unit/nva/model/Funding.java:20]: Quick question: The funding itself has an @id, which entails that every property relates to the object dereferenced by the @id. Is this the case, such that the @id always and only has the same values for e.g. funding amount, active*?

File: nva-datamodel-java/src/test/java/no/unit/nva/OntologyTest.java [hunk 2]
```diff
@@ -64,18 +66,27 @@ void shouldContainDistinctDescriptions() {
...
+    @ParameterizedTest
+    @MethodSource("provideModelPropertiesArguments")
+    void shouldContainEveryVisiblePropertyOfModel(String modelProperty) {
         var ontologyProperties = extractPropertiesFromOntology();
-        var modelProperties = new ArrayList<>(getModelProperties()).toArray(String[]::new);
-        assertThat(ontologyProperties, hasItems(modelProperties));
+        assertThat(ontologyProperties, hasItem(modelProperty));
+    }
+
+    public static Stream<Arguments> provideModelPropertiesArguments() {
```
Comment (brinxmat) [inline nva-datamodel-java/src/test/java/no/unit/nva/OntologyTest.java:84]: Call this modelPropertyProvider and keep it at the top of the class — this convention should probably be enforced by some rule, but isn't, alas.

File: nva-datamodel-java/src/test/java/no/unit/nva/OntologyTest.java [hunk 3]
```diff
@@ -64,18 +66,27 @@ void shouldContainDistinctDescriptions() {
         assertThat(duplicatesMessage, distinctValues, is(equalTo(ontologyValues)));
     }
 
-    @Test
-    void shouldContainEveryVisibleClassOfModel() {
+    @ParameterizedTest
+    @MethodSource("provideModelClassesArguments")
```
Comment (brinxmat) [inline nva-datamodel-java/src/test/java/no/unit/nva/OntologyTest.java:70]: Yep, this is a more constructive way of doing this :) call the method modelClassProvider.

File: nva-datamodel-java/src/test/java/no/unit/nva/OntologyTest.java [hunk 4]
```diff
@@ -64,18 +66,27 @@ void shouldContainDistinctDescriptions() {
...
-        var modelProperties = new ArrayList<>(getModelProperties()).toArray(String[]::new);
-        assertThat(ontologyProperties, hasItems(modelProperties));
+        assertThat(ontologyProperties, hasItem(modelProperty));
+    }
+
+    public static Stream<Arguments> provideModelPropertiesArguments() {
+        return Stream.of(Arguments.of(getModelProperties().toArray()));
+    }
+
+    public static Stream<Arguments> provideModelClassesArguments() {
```
Comment (brinxmat) [inline nva-datamodel-java/src/test/java/no/unit/nva/OntologyTest.java:88]: Also this should conventionally be at the top of the class.

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: nva-datamodel-java/src/main/java/no/unit/nva/model/Funding.java (hunk 1 above)
Comment (ketilaa) [reply nva-datamodel-java/src/main/java/no/unit/nva/model/Funding.java:20]: Not really. If source is NFR, activeFrom and activeTo can be resolved by the `id`. fundingAmount is something that can be set per use of the funding, e.g. to indicate that a funding is split on different projects. Other verified funding sources we integrate to in the future might not even have `activeFrom` and `activeTo`.

Comment (ketilaa) [reply nva-datamodel-java/src/main/java/no/unit/nva/model/Funding.java:20]: I think that from publication api perspective, a funding should have either an `id` or a `name`. The rest of the fields should be optional. `name` is used when the data is entered manually in the form.

Comment (ketilaa) [reply nva-datamodel-java/src/main/java/no/unit/nva/model/Funding.java:20]: Another aspect worth mentioning is that I have added source URI to verified fundings, so the `source` is also given if an `id` is provided.

File: nva-datamodel-java/src/test/java/no/unit/nva/OntologyTest.java (hunk 3 above)
Comment (ketilaa) [reply nva-datamodel-java/src/test/java/no/unit/nva/OntologyTest.java:70]: fixed

File: nva-datamodel-java/src/test/java/no/unit/nva/OntologyTest.java (hunk 4 above)
Comment (ketilaa) [reply nva-datamodel-java/src/test/java/no/unit/nva/OntologyTest.java:88]: fixed

File: nva-datamodel-java/src/test/java/no/unit/nva/OntologyTest.java (hunk 3 above)
Comment (ketilaa) [reply nva-datamodel-java/src/test/java/no/unit/nva/OntologyTest.java]: And yes, it is easier to see what class/property is the problem this way.

File: nva-datamodel-java/src/test/java/no/unit/nva/OntologyTest.java (hunk 2 above)
Comment (ketilaa) [reply nva-datamodel-java/src/test/java/no/unit/nva/OntologyTest.java]: fixed

File: nva-datamodel-java/src/main/java/no/unit/nva/model/Funding.java (hunk 1 above)
Comment (ketilaa) [reply nva-datamodel-java/src/main/java/no/unit/nva/model/Funding.java:20]: @brinxmat Do we want to change something based on this insight?

**Pull request merged**